coalib/output/ConsoleInteraction.py

```diff
@@ -225,12 +227,53 @@ def print_result(console_printer,
     console_printer.print(format_lines(result.message), delimiter="\n")
 
     if interactive:
+        cli_actions = CLI_ACTIONS
+        show_patch_action = ShowPatchAction()
+        if show_patch_action.is_applicable(result, file_dict, file_diff_dict):
+            diff_size = sum(len(diff) for diff in result.diffs.values())
+            if diff_size <= DIFF_EXCERPT_MAX_SIZE:
+                if section:
+                    show_patch_action.apply_from_section(result,
+                                                         file_dict,
+                                                         file_diff_dict,
+                                                         section)
+                cli_actions = tuple(action for action in cli_actions
+                                    if not isinstance(action, ShowPatchAction))
+            else:
+                print_diffs_info(result.diffs, file_dict, file_diff_dict,
+                                 console_printer)
         acquire_actions_and_apply(console_printer,
                                   log_printer,
                                   section,
                                   file_diff_dict,
                                   result,
-                                  file_dict)
+                                  file_dict,
+                                  cli_actions)
+
+
+def print_diffs_info(diffs, original_file_dict, file_diff_dict, printer):
+    for filename, diff in sorted(diffs.items()):
+        original_file = original_file_dict[filename]
+        if filename in file_diff_dict:
+            current_file = file_diff_dict[filename].modified
+            new_file = (file_diff_dict[filename] + diff).modified
+        else:
+            current_file = original_file
+            new_file = diff.modified
```

Comment:
I'm pretty sure this is duplication and should go into a function in the ShowPatchAction or so?